library/std/src/f32.rs

```diff
@@ -206,8 +206,11 @@ impl f32 {
     /// Fused multiply-add. Computes `(self * a) + b` with only one rounding
     /// error, yielding a more accurate result than an unfused multiply-add.
     ///
-    /// Using `mul_add` can be more performant than an unfused multiply-add if
-    /// the target architecture has a dedicated `fma` CPU instruction.
+    /// Using `mul_add` *can* be more performant than an unfused multiply-add if
+    /// the target architecture has a dedicated `fma` CPU instruction. However,
+    /// this is not always true, and care must be taken not to overload the
+    /// architecture's available FMA units when using many FMA instructions
+    /// in a row, which can cause a stall and performance degradation.
```

Comment:
This seems a little too specific, and may lead people to overgeneralize. This is highly likely to be CPU-dependent and model-dependent, and I wouldn't want to see someone thinking they should manually attempt to limit the number of FMA operations they do. (This is also something better left to a compiler's instruction scheduler, if anything.)

Having a warning here seems reasonable, but could it be generalized a little to just suggest that it "may be more performant", and that "the performance of FMA depends on the architecture and CPU, and may depend on the number of FMA operations versus non-FMA operations performed in a row"?